flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/writer/RecordWriterTest.java

```diff
@@ -524,6 +615,31 @@ public void read(DataInputView in) throws IOException {
 		}
 	}
 
+	/**
+	 * Broadcast channel selector that selects all the output channels.
+	 */
+	private static class Broadcast<T extends IOReadableWritable> implements ChannelSelector<T> {
+
+		private int[] returnChannel;
+		boolean set;
```

Comment:
Actually, this is a copy of `org.apache.flink.streaming.runtime.partitioner.BroadcastPartitioner` which is in the `flink-streaming-java` submodule, though.
And in general it is good to cache this rather than building a new array for every record...

Using `returnChannel.length == numberOfOutputChannels` makes sense though - @zhijiangW can you also create a hotfix commit changing this in `org.apache.flink.streaming.runtime.partitioner.BroadcastPartitioner`?

Reply:
yes, i just copy the code from `BroadcastPartitioner`, and i will simple this code in a hotfix commit later.